jfr/src/main/java/io/netty/handler/jfr/JfrEventHandler.java

```diff
@@ -0,0 +1,375 @@
+    public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
+        ChannelUnregisteredEvent event = new ChannelUnregisteredEvent();
+        event.begin();
+        ctx.fireChannelUnregistered();
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.commit();
+        }
+    }
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+        ChannelActiveEvent event = new ChannelActiveEvent();
+        event.begin();
+        ctx.fireChannelActive();
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.commit();
+        }
+    }
+
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        ChannelInactiveEvent event = new ChannelInactiveEvent();
+        event.begin();
+        ctx.fireChannelInactive();
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.commit();
+        }
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+        ExceptionCaughtEvent event = new ExceptionCaughtEvent();
+        event.begin();
+        ctx.fireExceptionCaught(cause);
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.commit();
+        }
+    }
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        UserEventTriggeredEvent event = new UserEventTriggeredEvent();
+        event.begin();
+        ctx.fireUserEventTriggered(evt);
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.userEvent = String.valueOf(evt);
+            event.commit();
+        }
+    }
+
+    @Override
+    public void bind(ChannelHandlerContext ctx, SocketAddress localAddress, ChannelPromise promise) throws Exception {
+        BindEvent event = new BindEvent();
+        event.begin();
+        ctx.bind(localAddress, promise);
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.localAddress = String.valueOf(localAddress);
+            event.commit();
+        }
+    }
+
+    @Override
+    public void connect(
+            ChannelHandlerContext ctx,
+            SocketAddress remoteAddress, SocketAddress localAddress, ChannelPromise promise) throws Exception {
+        ConnectEvent event = new ConnectEvent();
+        event.begin();
+        ctx.connect(remoteAddress, localAddress, promise);
+        event.end();
+        if (event.shouldCommit()) {
+            event.setChannel(ctx.channel());
+            event.remoteAddress = String.valueOf(remoteAddress);
+            event.localAddress = String.valueOf(localAddress);
```

Comment:
style nit: imho we should expose everything as `set*` or nothing at all. Mixing different styles is odd imho.